zk_toolbox/crates/zk_inception/src/commands/portal.rs

```diff
@@ -0,0 +1,178 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use anyhow::{anyhow, Context};
+use common::{docker, ethereum, ethereum::TokenInfo, logger};
+use config::{portal::*, traits::SaveConfig, ChainConfig, EcosystemConfig};
+use ethers::types::Address;
+use types::{BaseToken, L1Network};
+use xshell::Shell;
+
+use crate::{
+    commands::args::PortalArgs,
+    consts::{L2_BASE_TOKEN_ADDRESS, PORTAL_CONFIG_FILE, PORTAL_DOCKER_IMAGE, PORTAL_DOCKER_PORT},
+    messages::{
+        msg_portal_chain_not_initialized, msg_portal_failed_to_configure_chain,
+        msg_portal_starting_on, MSG_FAILED_TO_START_PORTAL_ERR,
+    },
+};
+
+pub async fn create_hyperchain_config(
+    chain_config: &ChainConfig,
+) -> anyhow::Result<HyperchainConfig> {
+    // Get L2 RPC URL from general config
+    let general_config = chain_config.get_general_config()?;
+    let rpc_url = &general_config
+        .api_config
+        .as_ref()
+        .context("api_config")?
+        .web3_json_rpc
+        .http_url;
+    // Check if L1 network is public
+    let public_l1_network_id = match chain_config.l1_network {
+        L1Network::Sepolia | L1Network::Mainnet => Some(chain_config.l1_network.chain_id()),
+        _ => None,
+    };
+    let secrets_config = chain_config.get_secrets_config()?;
+    let l1_rpc_url = secrets_config
+        .l1
+        .as_ref()
+        .context("l1")?
+        .l1_rpc_url
+        .expose_str();
+    let l1_network = if public_l1_network_id.is_none() {
+        // Initialize non-public L1 network
+        Some(L1NetworkConfig {
+            id: chain_config.l1_network.chain_id(),
+            name: chain_config.l1_network.to_string(),
+            network: chain_config.l1_network.to_string().to_lowercase(),
+            native_currency: NativeCurrency {
+                name: "Ether".to_string(),
+                symbol: "ETH".to_string(),
+                decimals: 18,
+            },
```

Comment:
I'd prefer to have it as a method for native currency. and what is the difference between `TokenInfo` and `NativeCurrency`?

Reply:
replaced NativeCurrency with TokenInfo